src/frontend/apps/impress/src/features/docs/doc-management/components/DocVisibility.tsx

```diff
@@ -44,55 +58,83 @@ export const DocVisibility = ({ doc }: DocVisibilityProps) => {
       $align="center"
       $justify="space-between"
       $gap="1rem"
+      $wrap="wrap"
     >
-      <IconBG iconName="public" $margin="none" />
+      <IconBG iconName="public" />
       <Box
-        $width="100%"
         $wrap="wrap"
         $gap="1rem"
-        $justify="space-between"
         $direction="row"
+        $align="center"
+        $flex="1"
+        $css={`
+          & .c__field__footer .c__field__text {
+            ${!doc.abilities.link_configuration && `color: ${colorsTokens()['greyscale-400']};`};
+          }
+        `}
       >
-        <Box $direction="row" $gap="1rem" $align="center">
-          <Switch
-            label={t(docPublic ? 'Doc public' : 'Doc private')}
-            defaultChecked={docPublic}
-            onChange={() => {
+        <Box $shrink="0" $flex="auto" $maxWidth="20rem">
+          <Select
+            label={t('Visibility')}
+            options={Object.values(LinkReach).map((linkReach) => ({
+              label: transLinkReach[linkReach].label,
+              value: linkReach,
+            }))}
+            onChange={(evt) =>
               api.mutate({
+                link_reach: evt.target.value as LinkReach,
                 id: doc.id,
-                link_reach: docPublic ? LinkReach.RESTRICTED : LinkReach.PUBLIC,
-                link_role: 'reader',
-              });
-              setDocPublic(!docPublic);
-            }}
-            disabled={!doc.abilities.link_configuration}
-            text={
-              docPublic
-                ? t('Anyone on the internet with the link can view')
-                : t('Only for people with access')
+              })
             }
+            value={doc.link_reach}
+            clearable={false}
+            text={transLinkReach[doc.link_reach].description}
+            disabled={!doc.abilities.link_configuration}
           />
         </Box>
-        <Button
-          onClick={() => {
-            navigator.clipboard
-              .writeText(window.location.href)
-              .then(() => {
-                toast(t('Link Copied !'), VariantType.SUCCESS, {
-                  duration: 3000,
-                });
-              })
-              .catch(() => {
-                toast(t('Failed to copy link'), VariantType.ERROR, {
-                  duration: 3000,
-                });
-              });
-          }}
-          color="primary"
-          icon={<span className="material-icons">copy</span>}
-        >
-          {t('Copy link')}
-        </Button>
+        {(doc.link_reach === LinkReach.AUTHENTICATED ||
+          doc.link_reach === LinkReach.PUBLIC) && (
```

Comment:
what do you think of creating a constant with explicit naming? like for example : 

`const showRestrictedOptions = doc.link_reach === LinkReach.AUTHENTICATED || doc.link_reach === LinkReach.PUBLIC`

or 

`const showRestrictedOptions = [LinkReach.AUTHENTICATED, LinkReach.PUBLIC].includes(doc.link_reach);`